store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/operation/FederatedOperation.java

```diff
@@ -16,22 +16,426 @@
 
 package uk.gov.gchq.gaffer.federatedstore.operation;
 
+import com.fasterxml.jackson.annotation.JsonGetter;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.core.type.TypeReference;
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+import org.apache.commons.lang3.exception.CloneFailedException;
+
+import uk.gov.gchq.gaffer.commonutil.Required;
+import uk.gov.gchq.gaffer.core.exception.GafferRuntimeException;
+import uk.gov.gchq.gaffer.exception.SerialisationException;
+import uk.gov.gchq.gaffer.federatedstore.util.FederatedStoreUtil;
+import uk.gov.gchq.gaffer.jsonserialisation.JSONSerialiser;
 import uk.gov.gchq.gaffer.operation.Operation;
-import uk.gov.gchq.gaffer.operation.OperationChain;
+import uk.gov.gchq.gaffer.operation.io.Input;
+import uk.gov.gchq.gaffer.operation.io.InputOutput;
+import uk.gov.gchq.gaffer.operation.io.Output;
+import uk.gov.gchq.gaffer.operation.serialisation.TypeReferenceImpl;
+import uk.gov.gchq.gaffer.store.operation.handler.util.OperationHandlerUtil;
+import uk.gov.gchq.koryphe.Since;
+import uk.gov.gchq.koryphe.Summary;
+import uk.gov.gchq.koryphe.ValidationResult;
+
+import java.lang.reflect.Field;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+
+import static java.util.Objects.isNull;
+import static java.util.Objects.nonNull;
+import static uk.gov.gchq.gaffer.federatedstore.FederatedStoreConstants.DEFAULT_SKIP_FAILED_FEDERATED_EXECUTION;
 
 /**
- * A FederatedOperation is an {@link FederatedOperation} that is a special
- * {@link uk.gov.gchq.gaffer.federatedstore.FederatedStore} operation and should always be handled by the
- * {@link uk.gov.gchq.gaffer.federatedstore.FederatedStore}.
+ * This operation federates a payload operation across a given set of graphs and merges the results with a given function.
+ *
+ * @param <INPUT>  Input type of the payload operation
+ * @param <OUTPUT> Output type of the merge function
  */
-public interface FederatedOperation extends Operation {
-    static boolean hasFederatedOperations(final OperationChain<?> operationChain) {
-        for (final Operation operation : operationChain.getOperations()) {
-            if (operation instanceof FederatedOperation) {
-                return true;
+@JsonPropertyOrder(value = {"class", "operation", "mergeFunction", "graphIds", "skipFailedFederatedExecution"}, alphabetic = true)
+@Since("2.0.0")
+@Summary("Federates a payload operation across given graphs and merges the results with a given function.")
+public class FederatedOperation<INPUT, OUTPUT> implements IFederationOperation, IFederatedOperation, InputOutput<INPUT, OUTPUT> {
+    private String graphIdsCsv;
```

Comment:
request to make this a list.

Reply:
request review "graph1,graph2,apples" or graph you can't see.
double check that apples is ignored.